### ISSUE TITLE
Upload a quay image on every master checkout

### ISSUE THREAD

**Issue by jcchavezs** [openzipkin/docker-zipkin#170]
As of now, docker images are being created on every [zipkin release](https://github.com/openzipkin/zipkin/releases) but sometimes, there is a need to use new changes without expecting a release. Currently it is possible by using `java -jar` but that is in java ecosystems, having the possibility to use docker would unblock non-java infrastructures to use the latest version in a docker container.

My proposal is that we add a hook on `openzipkin/zipkin` so on every merge into master we trigger a build on this repository and upload the image as `zipkin:master`. The reason for using `master` and not `latest` is to not to mess up with people using already `latest` as source of truth for getting the last release.

Opinions? @openzipkin/core @adriancole @abesto

**Comment by adriancole** [comment]
cc @openzipkin/devops-tooling

**Comment by abesto** [comment]
I don't see a down-side, especially with the consideration of not breaking `latest` in there. If there's a use-case here, I'd say go for it.

(What was the situation that triggered this? I have a weak feeling we may have an [X-Y Problem](http://mywiki.wooledge.org/XyProblem) here)

**Comment by jcchavezs** [comment]
So recently, some brilliant changes were introduced in the zipkin UI. Since they are in master and not in a release there is no way for people in non java ecosystems to use master. In my case, we use Zipkin in a docker container so it would be great if we could get fresh new changes from master in our docker setup same as java ecosystems would do (fair enough). I can do the changes my self but wanted to get some discussions around it to validate my idea.

**Comment by adriancole** [comment]
by non-java ecosystems, you mean those unwilling to run java, yet willing
to run docker to run java :)

On Sun, Jan 7, 2018 at 2:37 AM, José Carlos Chávez <notifications@github.com

**Comment by jcchavezs** [comment]
Related information: https://github.com/openzipkin/docker-zipkin/blob/master/RELEASE.md

**Comment by anuraaga** [comment]
We're now building an image on docker hub for every master push with the `master` tag so think this is solved.

**Comment by anuraaga** [comment]
Nevermind :) I think we'd need to tweak the Dockerfile to be able to build with the master source code. Now that we have the build infra can look into that.

**Comment by anuraaga** [comment]
We're now building every master commit to `openzipkin/zipkin` to the `openzipkin/zipkin:master` docker image tag, so now this should be ready.
